### PR TITLE
Ratings: add the structure for the give rating screen

### LINKED ISSUE #1880: Add basic UI and foundations

### PR DESCRIPTION
**Pull request by leandroalonso**
Adds the initial structure for the give rating screen:

1. Feature flag
2. Screen and view model

## To test

1. Run the app
2. Go to any podcast
3. Tap the stars
4. ✅ Nothing should happen
5. Go to Profile > Settings > Beta Features > enable `giveRatings`
6. Go back to the podcast
7. Tap the stars
5. ✅ A screen should appear
6. Tap dismiss
7. ✅ It should dismiss

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/FeatureFlag.swift [hunk 1]
```diff
@@ -46,42 +46,51 @@ enum FeatureFlag: String, CaseIterable {
...
-            return true
+            true
         case .newShowNotesEndpoint:
-            return true
+            true
         case .episodeFeedArtwork:
-            return Self.isTestFlight ? true : false
+            Self.isTestFlight ? true : false
+        case .giveRatings:
+            false
```
Comment (emilylaguna) [inline podcasts/FeatureFlag.swift]: Since we're touching this, what about adding a default case for all the true items, so we can reduce the list a bit?

**Review by emilylaguna — APPROVED** [review]
Tested on small screens, iPad, and a real device across iOS 15 to 17. All work. 

Left a non-blocking comment about the feature flags.

File: podcasts/FeatureFlag.swift (hunk 1 above)
Comment (leandroalonso) [reply podcasts/FeatureFlag.swift]: I wonder if this list should be there (and big) to bother us. There are a lot of flags there that need to be deleted.

But I don't have strong feelings against doing that — I have a small concern we might mess up and enable something by mistake but that shouldn't really happen.

Comment (emilylaguna) [reply podcasts/FeatureFlag.swift]: Makes sense. Lets leave it for now.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/33">7.50</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**